### PR TITLE
Remove ambiguities from TASTy reflect interface

### PR DESCRIPTION
**Pull request by nicolasstucki**
To avoid awkward imports like in https://github.com/lampepfl/dotty/pull/6042/files#diff-c60b7bdb36081577552ac360877d74ebR49

### CODE REVIEW

File: library/src/scala/tasty/reflect/Kernel.scala [hunk 1]
```diff
@@ -37,19 +37,19 @@ package scala.tasty.reflect
...
+ *           |               +- Project
+ *           |               +- Singleton
+ *           |               +- Refined
+ *           |               +- Applied
+ *           |               +- Annotated
+ *           |               +- MatchTypeTree
+ *           |               +- ByName
+ *           |               +- LambdaTypeTree
+ *           |               +- TypeBind
+ *           |               +- TypeBlock
```
Comment (liufengyun) [inline library/src/scala/tasty/reflect/Kernel.scala]: This refactoring reminds me of a disagreement among @xeno-by @odersky @olafurpg and me about _whether type trees and types_ should be the same.

Now opinions seem to converge that they should not be the same, as it's implemented in Tasty Reflect. However, I think one issue left unaddressed: _the confusion to meta programmers why there are seemingly overlapping Types and TypeTrees_ and usability problems evidenced by this PR.

To address the usability issue, I propose to just have one abstract type `TypeTree` (which could be used for positions) and remove all APIs to extract or construct `TypeTree`. Instead, meta-programmers should always work with `Types`, which is more reliable, as we do in the compiler.

Just a thought for discussion, no need to address in this PR.

File: library/src/scala/tasty/reflect/Kernel.scala [hunk 2]
```diff
@@ -550,137 +550,137 @@ trait Kernel {
...
+  def Singleton_ref(self: Singleton)(implicit ctx: Context): Term
 
-  def TypeTree_Singleton_apply(ref: Term)(implicit ctx: Context): TypeTree_Singleton
-  def TypeTree_Singleton_copy(original: TypeTree_Singleton)(ref: Term)(implicit ctx: Context): TypeTree_Singleton
+  def Singleton_apply(ref: Term)(implicit ctx: Context): Singleton
+  def Singleton_copy(original: Singleton)(ref: Term)(implicit ctx: Context): Singleton
 
   /** Type tree representing a type refinement */
-  type TypeTree_Refined <: TypeTree
+  type Refined <: TypeTree
```
Comment (liufengyun) [inline library/src/scala/tasty/reflect/Kernel.scala]: It's called `Refinement` in Type, do we need some consistency here?

File: compiler/src/dotty/tools/dotc/tastyreflect/KernelImpl.scala [hunk 3]
```diff
@@ -649,192 +649,192 @@ class KernelImpl(val rootContext: core.Contexts.Context, val rootPosition: util.
...
-  def TypeTree_Refined_refinements(self: TypeTree_Refined)(implicit ctx: Context): List[Definition] = self.refinements
+  def Refined_tpt(self: Refined)(implicit ctx: Context): TypeTree = self.tpt
+  def Refined_refinements(self: Refined)(implicit ctx: Context): List[Definition] = self.refinements
 
-  def TypeTree_Refined_copy(original: TypeTree_Refined)(tpt: TypeTree, refinements: List[Definition])(implicit ctx: Context): TypeTree_Refined =
+  def Refined_copy(original: Refined)(tpt: TypeTree, refinements: List[Definition])(implicit ctx: Context): Refined =
     tpd.cpy.RefinedTypeTree(original)(tpt, refinements)
 
-  type TypeTree_Applied = tpd.AppliedTypeTree
+  type Applied = tpd.AppliedTypeTree
```
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/tastyreflect/KernelImpl.scala]: Will do this renaming after #6193 for simplicity

File: library/src/scala/tasty/reflect/Kernel.scala (hunk 2 above)
Comment (nicolasstucki) [reply library/src/scala/tasty/reflect/Kernel.scala]: Will do this renaming after #6193 for simplicity

**Pull request merged**